cmd/dcs-web/search/query-rewrite.go

```diff
@@ -2,11 +2,52 @@
 package search
 
 import (
-	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
 )
 
+var (
+	start = regexp.MustCompile(`(?i)^\s*(-?(?:filetype|package|pkg|path|file)):(\S+)\s+`)
+	end = regexp.MustCompile(`(?i)\s+(-?(?:filetype|package|pkg|path|file)):(\S+)\s*$`)
+)
+
+func parseFilters(query *url.Values, qstr string, atstart bool) string {
+	q := *query
+
+	var re *regexp.Regexp
+	if atstart {
+		re = start
+	} else {
+		re = end
+	}
+
+	matches := re.FindStringSubmatchIndex(qstr)
+	for matches != nil {
+		// matches is [token_start, token_end, type_start, type_end, value_start, value_end]
```

Comment:
I use the indexes to know what part of the query string to remove, since we need to pull the filters out of the query string as we process them.  If I'm not getting the indexes from `re.FindStringSubmatchIndex`, then I would need to use some other means to know what to remove.

I figured `strings.TrimPrefix` would be clearer than getting the length of the match and using that to slice it out of the query string.